util/log/logrus.go

```diff
@@ -0,0 +1,50 @@
+package log
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/sirupsen/logrus"
+)
+
+type logrusEntryLogger struct {
+	entry logrus.FieldLogger
+}
+
+func (l logrusEntryLogger) Enabled() bool {
+	return true
+}
+
+func (l logrusEntryLogger) Info(msg string, keysAndValues ...interface{}) {
+	l.entry.Info(msg, keysAndValues)
+}
+
+func (l logrusEntryLogger) Error(err error, msg string, keysAndValues ...interface{}) {
+	l.entryWithKeysAndValues(keysAndValues).Error(err, msg)
+}
+
+func (l logrusEntryLogger) V(_ int) logr.Logger {
+	return l
+}
+
+func (l logrusEntryLogger) WithValues(keysAndValues ...interface{}) logr.Logger {
+	return NewLogrusLogger(l.entryWithKeysAndValues(keysAndValues))
+}
+
+func (l logrusEntryLogger) entryWithKeysAndValues(keysAndValues []interface{}) logrus.FieldLogger {
+	entry := l.entry
+	for i := 0; i < len(keysAndValues)-1; i += 2 {
+		key := fmt.Sprintf("%v", keysAndValues[i])
+		value := fmt.Sprintf("%v", keysAndValues[i+1])
+		entry = entry.WithField(key, value)
+	}
+	return entry
+}
+
+func (l logrusEntryLogger) WithName(_ string) logr.Logger {
+	return l
+}
+
+func NewLogrusLogger(fieldLogger logrus.FieldLogger) logr.Logger {
```

Comment:
Hey, you don't need to build your own adapter, there is one already :) https://github.com/bombsimon/logrusr

Reply:
thank you! used `https://github.com/bombsimon/logrusr` instead.